fuel-vm/src/tests/predicate.rs

```diff
@@ -292,8 +306,40 @@ async fn predicate() {
         op::ret(0x10),
     ];
 
-    assert!(execute_predicate(predicate.iter().copied(), expected_data, 0).await);
-    assert!(!execute_predicate(predicate.iter().copied(), wrong_data, 0).await);
+    assert!(
+        execute_predicate(
+            predicate.iter().copied(),
+            expected_data,
+            0,
+            NotSupportedEcal
+        )
+        .await
+    );
+    assert!(
+        !execute_predicate(predicate.iter().copied(), wrong_data, 0, NotSupportedEcal)
+            .await
+    );
+}
+
+#[tokio::test]
+async fn predicate_allows_ecal() {
```

Comment:
Are we missing a test showing that a predicate will fail if it using ecal and ecal isn't supported?

Also maybe a test that shows that the ecal is executed as well? `NoopEcal` shows that the vm won't panic, but it doesn't show that it's executing the ecal.

Reply:
Sure. Fixed in e5ba1d7d7.